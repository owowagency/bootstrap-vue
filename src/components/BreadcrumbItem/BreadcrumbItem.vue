<template>
    <li
        class="breadcrumb-item"
        :class="classes"
    >
        <!-- @slot Displays content inside the breadcrumb item -->
        <slot />
    </li>
</template>

<script lang="ts">
import {activeProps} from '@/composables/useActive';
import {computed} from 'vue';
import useActive from '@/composables/useActive';
import useClasses from '@/composables/useClasses';
</script>

<script lang="ts" setup>
const props = defineProps({
    ...activeProps,
});

const {classes} = useClasses(computed(() => [
    useActive(props.active).activeClass.value,
]));
</script>

<docs>
```vue
<BreadcrumbItem active>
    Products
</BreadcrumbItem>
```

## Imported Props

| Prop name | Description             | Type    | Values | Default |
| --------- | ----------------------- | ------- | ------ | ------- |
| active    | Adds the `active` class | boolean | -      | false   |
</docs>
