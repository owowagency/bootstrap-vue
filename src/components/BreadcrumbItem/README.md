# BreadcrumbItem

```vue
<BreadcrumbItem active>
    Products
</BreadcrumbItem>
```

## Imported Props

| Prop name | Description             | Type    | Values | Default |
| --------- | ----------------------- | ------- | ------ | ------- |
| active    | Adds the `active` class | boolean | -      | false   |

## Slots

| Name    | Description                                 | Bindings |
| ------- | ------------------------------------------- | -------- |
| default | Displays content inside the breadcrumb item |          |
