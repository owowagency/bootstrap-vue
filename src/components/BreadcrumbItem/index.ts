import BreadcrumbItem from './BreadcrumbItem.vue';

export default BreadcrumbItem;
