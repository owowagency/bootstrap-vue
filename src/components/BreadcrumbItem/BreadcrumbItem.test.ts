import BreadcrumbItem from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(BreadcrumbItem);

    componentSlotRenderTest(BreadcrumbItem);

    componentWrapperClassTest(BreadcrumbItem, {active: true}, 'active');
});
