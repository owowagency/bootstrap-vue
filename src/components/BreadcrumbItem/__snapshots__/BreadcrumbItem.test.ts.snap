// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<li class=\\"breadcrumb-item\\">
  <!-- @slot Displays content inside the breadcrumb item -->
</li>"
`;
